### PR TITLE
Ivanchuk/Rate provider and workshop

### PR DESCRIPTION
**Pull request by v-ivanchuk**
Implemented an ability to rate provider or workshop using an integer number from 1 to 5.

### CODE REVIEW

File: OutOfSchool/OutOfSchool.WebApi/Controllers/RatingController.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        /// Get all ratings from the database.
+        /// </summary>
+        /// <returns>List of all ratings.</returns>
+        [Authorize(Roles = "admin")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [HttpGet]
+        public async Task<IActionResult> Get()
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.WebApi/Controllers/RatingController.cs:43]: Will need to add pagination support from Yehor's PR later, as there will be a lot of records

File: OutOfSchool/OutOfSchool.WebApi/Services/WorkshopService.cs [hunk 2]
```diff
@@ -57,17 +66,29 @@ public async Task<IEnumerable<WorkshopDTO>> GetAll()
                 ? "Workshop table is empty."
                 : "Successfully got all records from the Workshop table.");
 
-            return workshops.Select(x => x.ToModel()).ToList();
+            var workshopsDTO = workshops.Select(x => x.ToModel()).ToList();
+
+            foreach (var workshop in workshopsDTO)
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.WebApi/Services/WorkshopService.cs:71]: same considerations as for provider :)

File: OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs [hunk 3]
```diff
@@ -63,7 +72,19 @@ public async Task<IEnumerable<ProviderDto>> GetAll()
                 ? "Provider table is empty."
                 : "Successfully got all records from the Provider table.");
 
-            return providers.Select(provider => provider.ToModel()).ToList();
+            var providersDTO = providers.Select(provider => provider.ToModel()).ToList();
+
+            foreach (var provider in providersDTO)
+            {
+                var providerRatings = await ratingRepository
```
[the remark this answers is not in the document]
Comment (v-ivanchuk) [reply OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs]: Dear @DmyMi,
These both points were done as for Provider and Workshop. Could you review one more time, please?
Thank you!

File: OutOfSchool/OutOfSchool.WebApi/Controllers/RatingController.cs (hunk 1 above)
Comment (v-ivanchuk) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/RatingController.cs:44]: Close as pagination will be added to all controllers that have "Get" method once Yehor finished with PR.

File: OutOfSchool/OutOfSchool.WebApi/Controllers/RatingController.cs [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        /// <param name="parentId">Id of Parent.</param>
+        /// <param name="entityId">Id of Entity.</param>
+        /// <param name="type">Entity type.</param>
+        /// <returns>Parent rating for the specified entity.</returns>
+        [Authorize(Roles = "parent,admin")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [HttpGet("{parentId}/{entityId}/{type}")]
```
[the remark this answers is not in the document]
Comment (v-ivanchuk) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/RatingController.cs]: Dear 
I have added two next variants for routing:
`example.com/rating/getparentrating/**provider**/{entityId}/parent/{parentId}` or
`example.com/rating/getparentrating/**workshop**/{entityId}/parent/{parentId}`
Please review the code.
Thank you!

**Review by DmyMi — APPROVED** [review]

File: OutOfSchool/OutOfSchool.WebApi/Controllers/RatingController.cs [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        public async Task<IActionResult> Delete(long id)
+        {
+            CheckIncomingId(id);
+
+            await service.Delete(id).ConfigureAwait(false);
+
+            return NoContent();
+        }
+
+        private void CheckIncomingId(long id)
```
Comment (dmitrykiev) [inline OutOfSchool/OutOfSchool.WebApi/Controllers/RatingController.cs:184]: please name it ValidateId like in other controllers

Comment (v-ivanchuk) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/RatingController.cs]: Dear @dmitryy90, this was done.

**Review by dmitrykiev — APPROVED** [review]

**Review by P0linux — APPROVED** [review]

**Pull request merged**